File: js/examples/zcsv_status.ts
import { ZippedCsv } from "../src/index";

const args = process.argv.slice(2);
if (args.length < 1) {
  console.error("Usage: tsx examples/zcsv_status.ts <path_to_zipped_csv_file>");
  process.exit(1);
}

try {
  const zcsv = new ZippedCsv(args[0]);
  const metadata = zcsv.getMetadata();
  const tables = zcsv.getTableNames();

  console.log(`File: ${args[0]}`);
  console.log(`Title: ${metadata.title || "(none)"}`);
  console.log(`Description: ${metadata.description || "(none)"}`);
  console.log(`Author: ${metadata.author || "(none)"}`);
  console.log(`Created: ${metadata.createdAt || "(none)"}`);
  console.log(`Updated: ${metadata.updatedAt || "(none)"}`);
  console.log(`Delimiter: ${JSON.stringify(metadata.delimiter)}`);
  console.log(`Quote char: ${JSON.stringify(metadata.quoteChar)}`);
  console.log(`Has header: ${metadata.hasHeader}`);
  console.log(`Tables (${tables.length}):`);
  for (const name of tables) {
    const csv = zcsv.getCsv(name);
    console.log(`  ${name}: ${csv.totalRows()} row(s), ${csv.totalColumns()} column(s)`);
  }
} catch (err) {
  console.error(`Error: ${err}`);
  process.exit(1);
}

File: cpp/examples/zcsv_status.cpp
// zcsv_status — prints metadata and table summary of a .zcsv archive.
//
// Usage: zcsv_status <archive.zcsv>

#include <cstdlib>
#include <iostream>
#include <string>

#include "zippedcsv.h"

static void PrintField(const std::string& label, const std::string& value) {
  if (!value.empty()) std::cout << "  " << label << ": " << value << "\n";
}

int main(int argc, char* argv[]) {
  if (argc < 2) {
    std::cerr << "Usage: zcsv_status <archive.zcsv>\n";
    return EXIT_FAILURE;
  }

  std::string path(argv[1]);
  mc::ZippedCsv z(path);

  std::cout << "Archive: " << path << "\n";

  auto meta = z.GetMetadata();
  std::cout << "Metadata:\n";
  PrintField("Title", meta.title);
  PrintField("Description", meta.description);
  PrintField("Author", meta.author);
  PrintField("Created", meta.created_at);
  PrintField("Updated", meta.updated_at);
  std::cout << "  Delimiter: '" << meta.delimiter << "'\n";
  std::cout << "  QuoteChar: '" << meta.quote_char << "'\n";
  std::cout << "  HasHeader: " << (meta.has_header ? "true" : "false") << "\n";

  auto names = z.GetTableNames();
  std::cout << "\nTables (" << names.size() << "):\n";
  for (const auto& name : names) {
    auto& csv = z.GetCsv(name);
    std::cout << "  " << name << " — " << csv.TotalColumns()
              << " column(s), " << csv.TotalRows() << " row(s)\n";
  }

  return EXIT_SUCCESS;
}
